internal/pkg/cli/svc_show.go

```diff
@@ -38,47 +37,53 @@ type showSvcOpts struct {
 	w             io.Writer
 	store         store
 	describer     describer
-	ws            wsSvcReader
 	sel           configSelector
-	initDescriber func(bool) error // Overriden in tests.
+	initDescriber func() error // Overriden in tests.
 }
 
 func newShowSvcOpts(vars showSvcVars) (*showSvcOpts, error) {
 	ssmStore, err := config.NewStore()
 	if err != nil {
-		return nil, fmt.Errorf("connect to environment datastore: %w", err)
+		return nil, fmt.Errorf("connect to config store: %w", err)
 	}
-	ws, err := workspace.New()
+	deployStore, err := deploy.NewStore(ssmStore)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("connect to deploy store: %w", err)
 	}
 
 	opts := &showSvcOpts{
 		showSvcVars: vars,
 		store:       ssmStore,
-		ws:          ws,
 		w:           log.OutputWriter,
 		sel:         selector.NewConfigSelect(vars.prompt, ssmStore),
 	}
-	opts.initDescriber = func(enableResources bool) error {
+	opts.initDescriber = func() error {
```

Comment:
Is there a reason why this func is inside another func? Would this work better? There might be a one liner catch that I can not remember right now. 

```
if err = opts.initDesriber(); err != nil {
   return err
}
```

Reply:
Good question! So that is because there are some fields of the struct we need extra info (usually will be retrieved in Ask() part) to init. So we define functions like `initDescriber` and call it when we have enough info to init the `Describer`

Reply:
the one liner catch is a great callout!